### PR TITLE
add local recordio reader interface

### PR DESCRIPTION
**Pull request by gongweibao**
fix https://github.com/PaddlePaddle/Paddle/issues/2542

### CODE REVIEW

**Comment by helinwang** [comment]
CI failed because:
```text
04:39:55]	[Step 1/1] File "/paddle/python/paddle/v2/reader/tests/.test_env/local/lib/python2.7/site-packages/paddle/v2/reader/creator.py", line 68, in reader
[04:39:55]	[Step 1/1] f = recordio.reader(path)
[04:39:55]	[Step 1/1] NameError: global name 'recordio' is not defined
```
Probably need to add pip install RecordIO package into the build process.

File: python/paddle/v2/reader/creator.py [hunk 1]
```diff
@@ -55,3 +55,22 @@ def reader():
         f.close()
 
     return reader
+
+
+def RecordIO(path):
```
Comment (helinwang) [inline python/paddle/v2/reader/creator.py]: We are following PEP 8 naming convention, please see: https://stackoverflow.com/a/159745/852385

**Comment by helinwang** [comment]
Great! Just a reminder that we will use `paddle.v2.reader.creator.RecordIO` for both local and cloud training (do not need to do in this PR).
Our goal is to reuse the same reader for local or cloud training:
> PaddlePaddle提供专用的data reader creator，生成给定RecordIO文件对应的data reader。无论在本地还是在云端，reader的使用方式都是一致的

The training in cloud the trainers will talk with the master server, so need to integrate with this PR: https://github.com/PaddlePaddle/Paddle/pull/2551

File: python/paddle/v2/reader/creator.py (hunk 1 above)
Comment (gongweibao) [inline python/paddle/v2/reader/creator.py]: 我看到文档设计的是大写，所以。。
我改过来。Done

Comment (helinwang) [inline python/paddle/v2/reader/creator.py]: 不好意思！写文档的时候没注意。

**Pull request merged**